lib/active_model/serializer/singular_reflection.rb

```diff
@@ -2,6 +2,12 @@ module ActiveModel
   class Serializer
     # @api private
     class SingularReflection < Reflection
+      def initialize(*)
+        super
+        @class_name = options.fetch(:class_name, name.to_s.camelize)
```

Comment:
Make sense

I think I modeled it on JSONAPI::Resources... (which has a simpler structure as well)